### DISCUSSION TITLE
How to inject own context into resolvers?

### DISCUSSION THREAD

**Discussion by frankmayer** [jycouet/kitql#240] [Q&A] [answered]
- What: I would like to inject some data(objects) into the context (3rd argument) of a resolver from a central point (for example in `modules/__init/index.ts`). 

- Why: This way I can directly access db/data-objects in each type-resolver-file that I have, without having to import the db/data-objects from their respective files in each resolver-file.

Is there an easy way to achieve this? 
Thank you!

**Comment by jycouet** [comment, marked as answer]
I have to setup the second demo with this 👍
I want to put this together in the coming days.

**Reply by frankmayer** [reply]
Great! Is there any hint you could give me until then (only if it's easy. Don't lose your sleep over it :smile: )?

**Reply by jycouet** [reply]
I did a few updates, you should get the demo on your machine to see with: 
```bash
npx degit jycouet/kitql/examples/demo-init demo-init
```

You will be able to see 
1/ how to [set the context](https://github.com/jycouet/kitql/blob/main/examples/demo-init/src/lib/graphql/kitQLServer.ts#L9)
2/ how to [get it in resolvers](https://github.com/jycouet/kitql/blob/main/examples/demo-init/src/lib/modules/user/resolvers/_Query.ts#L9)

Let me know if you managed 👍

**Reply by frankmayer** [reply]
Hi @jycouet, I actually tested exactly what you are proposing, but I did not have the https://github.com/jycouet/kitql/blob/main/examples/demo-init/src/lib/graphql/kitQLServer.ts#L24
it was only `getContext` instead of `context: getContext`, so it wasn't accessible in the resolvers. Great that it works now!! 
Thank you :100:

**Comment by frankmayer** [comment]
Sorry, had a lot on my plate in the week. will check this and my issue (#234) also, probably tomorrow. Thank you very much :+1:
